### PR TITLE
Feature/better pillars

### PR DESCRIPTION
**Pull request by jgminer**

### CODE REVIEW

**Comment by jgminer** [comment]
@jjurado @jlee4689

File: client/model/pillarModel.js [hunk 1]
```diff
@@ -160,6 +161,15 @@ define( [
                 return true;
             });
 
+            // basically verbatim from https://coderwall.com/p/imkqoq/resize-textarea-to-fit-content
+            $(document).on( 'change click keyup keydown paste cut', '.resize', function (){
+                $(this).height(0).height(this.scrollHeight);
+            }).find( '.resize' ).change();
```
Comment (nicmoski) [inline client/model/pillarModel.js]: this doesn't have a size limit?

Comment (jgminer) [inline client/model/pillarModel.js]: no, what is your $0.02 on it? anything with resize in the class I want to resize, unless there is a smart way to only change the ones that are about to be displayed (which i tried to find)

Comment (jgminer) [inline client/model/pillarModel.js]: nevermind, just realized you meant height - no it doesn't

File: client/model/pillarModel.js [hunk 2]
```diff
@@ -185,10 +196,18 @@ define( [
...
+                            jsonlint.parse(pair.value);
                         } catch(err) {
-                            swal("Error", "Please make sure your values consist of valid JSON", 'error');
+                            if (err.name == 'TypeError') {
+                                err.message = "No value provided for key " + pair.key + ". Remove key-value pair if not " +
+                                "needed.";
+                            }
                             ret = false;
+                            // save the first error we saw so end user can tell which line is invalid
+                            if (first_error == "") {
```
Comment (nicmoski) [inline client/model/pillarModel.js]: what if it's not the first error?

Comment (jgminer) [inline client/model/pillarModel.js]: the idea behind this was to expose only one error to the user at a time, top-down. so if the first row has an error and so does the second, the first row error will be shown until fixed - and then the second will be shown.

**Pull request merged**